### ISSUE TITLE
Sites created with Reactivesearch have rendering issues on mobile

### ISSUE THREAD

**Issue by siddharthlatest** [appbaseio/reactivesearch#211]
**Issue Type:**

Possible Bug

**Description:**

Sites created with Reactivesearch have rendering issues when testing with https://search.google.com/test/mobile-friendly. It shows a blank screen.

**Screenshots:**

![screen shot 2018-01-16 at 6 25 32 am](https://user-images.githubusercontent.com/630042/34967011-d5c0cae4-fa85-11e7-9b0f-5df305d95520.png)

**Minimal reproduction of the problem with instructions:**

Go to https://search.google.com/test/mobile-friendly.

**Reactivesearch version:** x.y.z

2.0.0

**Comment by divyanshu013** [comment]
IMO this could be an issue with transpilation. We're targeting only latest 2 versions of browsers. @metagrover anyway we can test this hypothesis?

**Comment by Guazi** [comment]
Googlebot also has trouble crawling reactivesearch websites, when using Fetch as Google through the webmaster tools (same result using both Desktop and Mobile render):

![selection_078](https://user-images.githubusercontent.com/3461028/35070050-174e0448-fbaa-11e7-812c-83ca96df93c2.jpg)

**Comment by metagrover** [comment]
The issue is with the polyfill being absent. Google bot uses an older version of Chrome to render pages. Adding the following [polyfill](https://polyfill.io/v2/docs/) fixes the issue:

```
<script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
```

![screen shot 2018-01-18 at 2 48 30 pm](https://user-images.githubusercontent.com/9116042/35090241-119788fc-fc5f-11e7-94d9-9dc5b0f6b159.png)

**Comment by divyanshu013** [comment]
We should add polyfill in
- [ ] website
- [ ] demo apps

**Comment by Guazi** [comment]
I added     <script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script> to the bottom of my index.html and it still won't render when using the Mobile Friendly tool or the Fetch as Google tool.  I cloned gitxplore-app and added it to index.html and still the same, won't render.

**Comment by siddharthlatest** [comment]
@metagrover can we verify if this works on a demo app?

**Comment by Guazi** [comment]
Looked into debugging this and came across this site:

https://blog.codaxy.com/debugging-googlebot-crawl-errors-for-javascript-applications-5d9134c06ee7

I added the code to my app and here's the stack trace:

![selection_088](https://user-images.githubusercontent.com/3461028/35201042-aedd657c-fee5-11e7-8e00-2d85710bd699.jpg)

**Comment by metagrover** [comment]
Thanks @Guazi, that was incredibly helpful. The problem was with the web-socket connection handshake, which I've fixed and tested with the booksearch app here:

![screen shot 2018-01-22 at 10 54 42 am](https://user-images.githubusercontent.com/9116042/35206577-94c7f624-ff63-11e7-9d1e-f4f4ee6716b7.png)


:tada: FUN FACT: I'm not using any polyfill script here, works perfectly without it.

**Comment by divyanshu013** [comment]
Probably because of babel polyfill?

**Comment by metagrover** [comment]
Yes, and that's why we should not remove babel-polyfill from appbase-js

**Comment by metagrover** [comment]
Fixed in 2.2.0 🎉
